examples/apiserver/apiserver.go

```diff
@@ -55,18 +55,20 @@ func newStorageFactory() genericapiserver.StorageFactory {
 }
 
 func NewServerRunOptions() *genericoptions.ServerRunOptions {
-	serverOptions := genericoptions.NewServerRunOptions().WithEtcdOptions()
-	serverOptions.InsecurePort = InsecurePort
+	serverOptions := genericoptions.NewServerRunOptions().WithEtcdOptions().WithSecureServingOptions().WithInsecureServingOptions()
```

Comment:
> Do we really want this pattern?

We might.  It could actually work for some of our downstream projects. Like I said in the etcd pull.  I see that as getting all the pieces ready and then making a choice.